src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -105,11 +108,17 @@ case class AvroRelation(
       .toStream
       .map(_.getPath)
       .flatMap(getAllFiles(fs)(_))
+    val isAvroFile : Path => Boolean =
+      if(hadoopConfiguration.getBoolean(IgnoreAvroFileExtensionProperty, true)) {
+        _.getName.endsWith("avro")
+      } else {
+        Function.const(true)
+      }
     val singleFile = statuses
```

Comment:
maybe something like this (to avoid creating this temporary function)
```scala
val singleFile = if (hadoopConfiguration.getBoolean(IgnoreAvroFileExtensionProperty, true)) {
  statuses.find(_.getName.endsWith("avro"))
} else {
  statuses.headOption
}.getOrElse(sys.error(s"Could not find .avro file with schema at $path"))
```
also, consider renaming IgnoreAvroFileExtensionProperty, because that reads like the opposite of what it is